src/components/Calendar/src/Calendar.vue

```diff
@@ -0,0 +1,448 @@
+<template>
+	<div
+		:class="$s.Calendar"
+		v-bind="$attrs"
+		v-on="$listeners"
+	>
+		<div :class="$s.CalendarHeader">
+			<m-button
+				:disabled="shouldDisableHeaderButton(-1)"
+				:color="headerButtonColor"
+				size="small"
+				variant="primary"
+				tabindex="-1"
+				@click="incMonth(-1)"
+			>
+				<chevron-left
+					:class="$s.CalendarHeaderButtonIcon"
+				/>
+			</m-button>
+
+			<span :class="$s.CalendarHeaderTitle">
+				{{ monthName }}
+			</span>
+
+			<m-button
+				:disabled="shouldDisableHeaderButton(1)"
+				size="small"
+				variant="primary"
+				tabindex="-1"
+				color="#f2f2f2"
+				@click="incMonth(1)"
+			>
+				<chevron-right
+					:class="$s.CalendarHeaderButtonIcon"
+				/>
+			</m-button>
+		</div>
+		<table :class="$s.CalendarTable">
+			<thead>
+				<tr>
+					<th
+						v-for="day in weekdays"
+						:key="`day-${day}`"
+						:class="$s.DateHeaderCell"
+					>
+						{{ day }}
+					</th>
+				</tr>
+			</thead>
+			<tbody>
+				<tr
+					v-for="(week, wi) in weeks"
+					:key="`week-${wi}`"
+				>
+					<td
+						v-for="(date, di) in week"
+						:key="`date-${wi-di}`"
+						:class="$s.DateCell"
+						@click.prevent="emitDate(date)"
+					>
+						<button
+							v-if="date"
+							:class="[$s.DateCellButton, {
+								[$s.selected]: isDateSelected(date),
+								[$s.disabled]: isDateDisabled(date),
+								[$s.today]: isToday(date)
+							}]"
+							type="button"
+							tabindex="-1"
+						>
+							{{ date }}
+						</button>
+					</td>
+				</tr>
+			</tbody>
+		</table>
+	</div>
+</template>
+
+<script>
+import { MButton } from '@square/maker/components/Button';
+import ChevronLeft from '@square/maker-icons/ChevronLeft';
+import ChevronRight from '@square/maker-icons/ChevronRight';
+
+const isDateFormat = (string) => !string || /^\d{4}-\d{2}-\d{2}$/.test(string); // YYYY-MM-DD
+
+/**
+ * @inheritAttrs div
+ * @inheritListeners div
+ */
+export default {
+	components: {
+		MButton,
+		ChevronLeft,
+		ChevronRight,
+	},
+
+	inheritAttrs: false,
+
+	model: {
+		prop: 'selectedDate',
+		event: 'calendar:update',
+	},
+
+	props: {
+		/**
+		 * Language specific format to use
+		 */
+		locale: {
+			type: String,
+			default: 'en-US',
+		},
+		/**
+		 * Selected date value (YYYY-MM-DD)
+		 */
+		selectedDate: {
+			type: String,
+			default: '',
+			validator: isDateFormat,
+		},
+		/**
+		 * Disable the dates before the min-date
+		 */
+		minDate: {
+			type: String,
+			default: '',
+			validator: isDateFormat,
+		},
+		/**
+		 * Disable the dates after the max-date
+		 */
+		maxDate: {
+			type: String,
+			default: '',
+			validator: isDateFormat,
+		},
+		/**
+		 * List of disabled dates
+		 */
+		disabledDates: {
+			type: Array,
+			default: () => ([]),
+			validator: (v) => v.every((date) => isDateFormat(date)),
+		},
+	},
+
+	data() {
+		return {
+			headerButtonColor: '#f2f2f2',
```

Comment:
How come this one is in `data` while color values are used directly in other cases?

Reply:
I can't use CSS var here b/c I'm passing a color to the button component

Reply:
I meant to ask why use `data()` at all instead of just `<m-button color="#f2f2f2"...`?